src/overlays/actors/ovl_En_Ice_Hono/z_en_ice_hono.c

```diff
@@ -21,37 +32,342 @@ const ActorInit En_Ice_Hono_InitVars = {
+    EnIceHono_SetupActionDroppedFlame(this);
+}
+
+void EnIceHono_InitSmallFlame(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChainSmallFlame);
+    this->actor.scale.x = this->actor.scale.z = this->actor.scale.y = 0.0008f;
+    this->actor.gravity = -0.3f;
+    this->actor.minVelocityY = -4.0f;
+    this->actor.shape.unk_08 = 0.0f;
+
+    EnIceHono_SetupActionSmallFlame(this);
+}
+
+void EnIceHono_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+    s16 params = this->actor.params;
+
+    switch (this->actor.params) {
+        case -1:
+            EnIceHono_InitCapturableFlame(&this->actor, globalCtx);
+            break;
+        case 0:
+            EnIceHono_InitDroppedFlame(&this->actor, globalCtx);
+            break;
+        case 1:
+        case 2:
+            EnIceHono_InitSmallFlame(&this->actor, globalCtx);
+            break;
+    }
+
+    if ((this->actor.params == -1) || (this->actor.params == 0)) {
+        Lights_PointNoGlowSetInfo(&this->lightInfo, this->actor.posRot.pos.x, (s16)this->actor.posRot.pos.y + 0x0A,
+                                  this->actor.posRot.pos.z, 155, 210, 255, 0);
+        this->lightNode = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfo);
+        this->unk_154 = Math_Rand_ZeroOne() * (0x1FFFF / 2.0f);
+        this->unk_156 = Math_Rand_ZeroOne() * (0x1FFFF / 2.0f);
+        // Translates to: "(ice flame)"
+        osSyncPrintf("(ice 炎)(arg_data 0x%04x)\n", this->actor.params);
+    }
+}
+
+void EnIceHono_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+
+    if ((this->actor.params == -1) || (this->actor.params == 0)) {
+        LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+}
+
+u32 EnIceHono_LinkCloseAndFacing(EnIceHono* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->actor.xzDistFromLink < 60.0f) {
+        Vec3f tempPos;
+        tempPos.x = Math_Sins(this->actor.yawTowardsLink + 0x8000) * 40.0f + player->actor.posRot.pos.x;
+        tempPos.y = player->actor.posRot.pos.y;
+        tempPos.z = Math_Coss(this->actor.yawTowardsLink + 0x8000) * 40.0f + player->actor.posRot.pos.z;
+        if (EnIceHono_SquareDist(&tempPos, &this->actor.posRot.pos) <= SQ(40.0f)) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void EnIceHono_SetupActionCapturableFlame(EnIceHono* this) {
+    this->actionFunc = EnIceHono_CapturableFlame;
+    this->alpha = 255;
+    this->actor.shape.unk_08 = -1000.0f;
+}
+
+void EnIceHono_CapturableFlame(EnIceHono* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx))
+        this->actor.parent = 0;
+    else if (EnIceHono_LinkCloseAndFacing(this, globalCtx))
+        func_8002F434(&this->actor, globalCtx, 0x7E, 60.0f, 100.0f);
```

Comment:
I think the general style is to use brackets everywhere `if (...) {...} else {...}`